EvolvSwiftSDK/EvolvSwiftSDK/Sources/Core/EvolvClientImpl.swift

```diff
@@ -248,8 +248,9 @@ final class EvolvClientImpl: EvolvClient {
     public func emit<T: Encodable>(eventType: String, metadata: T?, flush: Bool) {
         let event = EvolvCustomEventForSubmission(type: eventType, uid: options.participantID, metadata: metadata)
         
+        let oldEvents = evolvStore.evolvContext.events;
         evolvStore.evolvContext.events.append(.init(type: eventType, timestamp: Date()))
-        
+        self.evolvStore.evolvContext.contextChanged(key: "events", value: evolvStore.evolvContext.events, before: oldEvents)
```

Comment:
Is `evolvStore.evolvContext.events` passed by value or by reference? Or rather, is `evolvStore.evolvContext.events` treated as immutable once `contextChanged` receives it?

Reply:
From this it seemed that shallow copies here are by value
https://stackoverflow.com/questions/27812433/how-do-i-make-a-exact-duplicate-copy-of-an-array